record Proverb(String[] words) {

    String recite() {
        StringBuilder proverb = new StringBuilder();
        if(words.length == 0) {
            return "";
        }
        if(words.length == 1){
            return proverb.append("And all for the want of a ")
                    .append(words[0])
                    .append(".")
                    .toString();
        }
        for(int i = 0 ; i < words.length-1 ; i++) {
            proverb.append("For want of a ")
                    .append(words[i])
                    .append(" the ")
                    .append(words[i+1])
                    .append(" was lost.")
                    .append("\n");
        }
        return proverb.append("And all for the want of a ")
                .append(words[0])
                .append(".")
                .toString();
    }

}
